pkg/microservice/aslan/core/workflow/service/workflow/pipeline_validation.go

```diff
@@ -753,8 +754,35 @@ func getImageInfoFromWorkload(envName, productName, serviceName, container strin
 		}
 		return "", errors.New("no container in deployment found")
 	default:
-		log.Errorf("The type of workload is not supported: %s", serviceInfo.WorkloadType)
-		return "", errors.New("the type of workload is not supported")
+		// since in some version of the service, there are no workload type, we need to do something with this
+		selector := labels.Set{setting.ProductLabel: product.ProductName, setting.ServiceLabel: serviceName}.AsSelector()
+		var deployments []*appsv1.Deployment
+		deployments, err = getter.ListDeployments(product.Namespace, selector, kubeClient)
+		if err != nil {
+			return "", err
+		}
+		var statefulSets []*appsv1.StatefulSet
+		statefulSets, err = getter.ListStatefulSets(product.Namespace, selector, kubeClient)
+		if err != nil {
+			return "", err
+		}
+		for _, deploy := range deployments {
```

Comment:
Perhaps it would be better to check the first workload once, instead of checking all the workload in advance.